### ISSUE TITLE
Package listing is duplicated when using `-Wunused-packages`

### ISSUE THREAD

**Issue by guibou** [tweag/rules_haskell#1309]
**Describe the bug**

GHC 8.10 adds the `-Wunused-packages` which tells which packages are unused during a build. This is super useful. For example, I was able to remove many useless edges in the build graph of a client project, hence reducing build time thanks to improved paralellism.

However, with rules_haskell, the warnings are duplicated:```

```
<no location info>: error: [-Wunused-packages, -Werror=unused-packages]
    The following packages were specified via -package or -package-id flags,
    but were not needed for compilation:
      - containers-0.6.2.1
      - containers-0.6.2.1
```

When bare `ghc` call gives:

```
<no location info>: warning: [-Wunused-packages]
    The following packages were specified via -package or -package-id flags,
    but were not needed for compilation:
      - containers
```

**To Reproduce**

https://github.com/tweag/rules_haskell/commit/82ace43c74f098f67def7cbbb485f2bd737c5c26

is a commit with reproduction code on rules_haskell.

```
$ nix-shell
[nix-shell:~/tweag/rules_haskell]$ bazel build tests/unused_packages/...
INFO: Analyzed 3 targets (0 packages loaded, 0 targets configured).
INFO: Found 3 targets...
ERROR: /home/guillaume/tweag/rules_haskell/tests/unused_packages/BUILD.bazel:8:1: HaskellBuildBinary //tests/unused_packages:binary-simple failed (Exit 1) ghc_wrapper failed: error executing command bazel-out/host/bin/haskell/ghc_wrapper bazel-out/k8-fastbuild/bin/tests/unused_packages/compile_flags_binary-simple_HaskellBuildBinary ... (remaining 1 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox

<no location info>: error: [-Wunused-packages, -Werror=unused-packages]
    The following packages were specified via -package or -package-id flags,
    but were not needed for compilation:
      - containers-0.6.2.1
      - containers-0.6.2.1
INFO: Elapsed time: 0.597s, Critical Path: 0.46s
INFO: 0 processes.
FAILED: Build did NOT complete successfully

[nix-shell:~/tweag/rules_haskell]$ ghc^C

[nix-shell:~/tweag/rules_haskell]$ exit
$ nix-shell -p haskell.packages.ghc8101.ghc

[nix-shell:~/tweag/rules_haskell]$ ghc tests/unused_packages/Main.hs -fforce-recomp -package containers -Wunused-packages
[1 of 1] Compiling Main             ( tests/unused_packages/Main.hs, tests/unused_packages/Main.o )
Linking tests/unused_packages/Main ...

<no location info>: warning: [-Wunused-packages]
    The following packages were specified via -package or -package-id flags,
    but were not needed for compilation:
      - containers
```

**Expected behavior**

I would like no duplicated output.

**Environment**
 - OS name + version: Linux (Nixos)
 - Bazel version: 2.1.0- (@non-git)
 - Version of the rules: the previous commit on top of `76f7b1c7328f59bf2f38b56bce0036153c65a40d`

**Additional context**

**Comment by hanshoglund** [comment]
I minimized the GHC invocation generated by rules_haskell and and the problem persists. 

Upstreamed: https://gitlab.haskell.org/ghc/ghc/issues/18125

**Comment by mboes** [comment]
This is tracked upstream and there's nothing we need to do to adapt to the fix once it lands. Closing.
